### PR TITLE
github: add workflows for building and publishing kata artefacts on ppc64le

### PR DESCRIPTION
**Pull request by Amulyam24**
This is the first step of the migration of jobs from Jenkins to GHA. 

Fixes: #8458

### CODE REVIEW

**Comment by Amulyam24** [comment]
Hi @fidencio @stevenhorsman , can you please help me in adding a self hosted ppc64le runner?

**Comment by stevenhorsman** [comment]
> Hi @fidencio @stevenhorsman , can you please help me in adding a self hosted ppc64le runner?

Sure, send me a slack when you are ready and I can send through the token and instructions

File: .github/workflows/build-kata-static-tarball-ppc64le.yaml [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+
+      - name: Rebase atop of the latest target branch
+        run: |
+          ./tests/git-helper.sh "rebase-atop-of-the-latest-target-branch"
+        env:
+          TARGET_BRANCH: ${{ inputs.target-branch }}
+
+      - name: Build ${{ matrix.asset }}
+        run: |
+          CROSS_BUILD=true ARCH=ppc64le TARGET_ARCH=ppc64le make "${KATA_ASSET}-tarball"
```
Comment (BbolroC) [inline .github/workflows/build-kata-static-tarball-ppc64le.yaml:63]: Assuming that a runner is a ppc64le native machine, I was wondering if it is necessary.

File: tools/packaging/kata-deploy/local-build/dockerbuild/install_oras.sh [hunk 2]
```diff
@@ -29,9 +29,21 @@ fi
 
 arch=$(uname -m)
 if [ "${arch}" = "ppc64le" ]; then
-	echo "An ORAS release for ppc64le is not available yet."
-	exit 0
-fi
+ 	echo "Building oras from source"
+	go_version="go1.21.3"
```
Comment (BbolroC) [inline tools/packaging/kata-deploy/local-build/dockerbuild/install_oras.sh:33]: Why don't you reuse the existing [script](https://github.com/kata-containers/kata-containers/blob/main/ci/install_go.sh) for this?

**Review by BbolroC — COMMENTED** [review]
Thanks for the request. I've left 2 comments. 

Additionally, I've noticed that we finally have workflows to build artifacts for 4 different architectures where no steps including a `if: always()` condition for a self-hosted runner (running 24/7) are engaged, but to produce lots of duplicated codes. 

I was wondering if this PR would be a right place to incorporate the workflows or we could do it in a separate PR. @fidencio and @stevenhorsman

File: tools/packaging/kata-deploy/local-build/dockerbuild/install_oras.sh (hunk 2 above)
Comment (BbolroC) [reply tools/packaging/kata-deploy/local-build/dockerbuild/install_oras.sh:33]: Hmm... it fetches the `test` repo which is going to be deprecated soon. I think we have to rewrite the script in a separate PR and refactor this there. Just ignore this comment.

**Review by BbolroC — APPROVED** [review]
Other than the comment regarding the deduplication, LGTM! Thanks @Amulyam24 !

File: tools/packaging/kata-deploy/local-build/dockerbuild/install_oras.sh (hunk 2 above)
Comment (Amulyam24) [reply tools/packaging/kata-deploy/local-build/dockerbuild/install_oras.sh:33]: @BbolroC, this is a temporary fix to build and use Oras, once the release it out for ppc64le, I will switch to using the artefacts from releases.

**Review by stevenhorsman — APPROVED** [review]
It would be good to see a run on this testing on your fork, but I think the code looks okay and expected.

I definitely understand @BbolroC's comments about the duplication:
- The build-kata-static-tarball workflow only differs in artifact name, runner and assets that are created
- The publish workflows only varies on the artifact name and self-host runner prepare in ppc
- The release workflows are pretty identical other than the build workflows they call
I'd like to hear @fidencio's thought, but I think we should consider refactoring these to try and get a single workflow with passing the arch in. We might have a couple of steps like:
```
 - name: Prepare the self-hosted runner
        if: ${{ arch == 'ppc64le' }}
        run: ${HOME}/scripts/prepare_runner.sh)
```
but I think it's worth tying to see if it's not too much of a mess and would potentially save on the maintenance cost of updating 4 versions in future.
I'm okay for this to be done as a separate PR though.

**Comment by Amulyam24** [comment]
@stevenhorsman, testing done for both this and containerd PR, PTAL for reference - https://github.com/Amulyam24/kata-containers/actions/runs/6979218191/job/18992074251?pr=6

**Review by GabyCT — APPROVED** [review]

**Comment by Amulyam24** [comment]
Hi @stevenhorsman, we have 3 approvals, can this PR be merged?

**Comment by fidencio** [comment]
> I'm okay for this to be done as a separate PR though.

I am with Steve and Choi here, but I'd for sure prefer this addition coming on a different PR.

**Pull request merged**